### PR TITLE
fix(ft): chunk encode_texts forward pass by batch_size

### PR DESCRIPTION
**Pull request by Souptik96**
Addresses **item 2** of #244.

`encode_texts` tokenized and forwarded the **entire** `texts` list in a single pass, so peak memory scaled with `len(texts)` rather than with any batch size. `evaluate_embeddings` is an in-repo caller that makes this easy to hit: it hands `encode_texts` the full anchor, positive and negative lists, so evaluating a large set is three unbounded forward passes.

### Change

- `encode_texts` gained a `batch_size` parameter (default `32`) and now encodes in chunks, extending the result list per chunk.
- `evaluate_embeddings` gained the same parameter and threads it through to its three `encode_texts` calls.
- `batch_size < 1` raises `ValueError` rather than silently producing an empty result (`range(0, n, 0)` would raise an opaque error).

The new parameter is appended **last** in both signatures, so the existing positional call style used inside `evaluate_embeddings` (`encode_texts(anchors, model, projector, tokenizer, device, max_length)`) and in `tests/test_ft_embeddings.py` keeps working unchanged.

**Chunking is numerically transparent.** Padding is now per-batch rather than across the whole input, but `_mean_pooling` divides by the attention mask, so pad positions never contribute to the pooled vector. There is a test asserting this explicitly (`batch_size=1`, `batch_size=4` and one single pass all produce the same embeddings to `atol=1e-6`).

### Scope

#244 lists five items; this PR deliberately does only item 2, the one that is a straightforward bug. Not touched here:

- **item 1 (`trust_remote_code`)** — real, but it is a new feature rather than a broken path: the field does not exist on `EmbeddingConfig` at all, and would need forwarding to `AutoModel.from_pretrained`, `AutoTokenizer.from_pretrained` and `load_embedding_model`.
- **item 3 (`device` default)** — a `_default_device()` helper changes the default behaviour of every config, so it seemed worth keeping separate.
- **item 4 (`run_embedding_sft_mlx`)** — the issue itself offers "either remove it or port it to `mlx_lm`", which is a maintainer decision, and I have no Apple silicon to verify a port on.
- **item 5 (triplet JSONL loader)** — a new feature.

Happy to follow up on any of these if you'd like them, in whatever order suits you.

### Tests

Added `TestEncodeTextsBatching` (7 tests) to `tests/test_ft_embeddings.py`: chunk boundaries and call counts, numerical equivalence across batch sizes, a bounded default, empty input doing zero forward passes, `ValueError` on non-positive batch sizes, and `evaluate_embeddings` threading the value through.

These use a stub tokenizer plus a tiny `torch.nn.Embedding` model, so unlike the existing `@pytest.mark.slow` classes they need **no network and no HuggingFace download**. Token ids are derived from `sum(ord(c))` rather than `hash()` so they are stable across processes.

### Verification

Windows 11, Python 3.13, `torch 2.13.0+cpu`, in a throwaway venv with `pip install -e .`

```
# new tests
$ pytest tests/test_ft_embeddings.py -k Batching -q
7 passed, 18 deselected

# negative control: revert ONLY the embeddings.py hunk, keep the tests
$ git checkout HEAD -- npcpy/ft/embeddings.py && pytest tests/test_ft_embeddings.py -k Batching -q
7 failed, 18 deselected

# whole file, fast tests
pristine main : 10 passed, 8 deselected
this branch   : 17 passed, 8 deselected     (10 pre-existing + 7 new)

# pre-existing integration tests that actually call the changed functions
# (real sentence-transformers/all-MiniLM-L6-v2 download)
$ pytest tests/test_ft_embeddings.py::TestClassicalEmbeddingTraining -q
pristine main : 4 passed
this branch   : 4 passed
```

Lint, using the exact invocation from `.github/workflows/ci.yml`:

```
$ ruff check npcpy/ft/embeddings.py tests/test_ft_embeddings.py --select=E,F,W --ignore=E501,E402,F401
All checks passed!        # identical on pristine main
```

### Limitations, stated honestly

- `tests/test_ft_embeddings.py` is **not currently run by CI** (`ci.yml` enumerates test files individually and this one is not among them, presumably because it needs `torch`). So these tests will not gate anything until that file is added to a workflow — happy to add it if you want, though it would mean putting `torch` in the CI install.
- I verified on **CPU only**. The chunking is device-agnostic, but I have no CUDA or MPS hardware to confirm the memory win on an accelerator.
- The MLX paths were not exercised at all.

### CODE REVIEW

**Comment by cagostino** [comment]
looks good, thanks for picking this item up!
